### PR TITLE
US854020: Improve the poison message for Workers using the Document Worker Framework

### PR DESCRIPTION
**Pull request by andyreidz**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=854020

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -101,7 +101,7 @@
             <dependency>
                 <groupId>com.github.workerframework</groupId>
                 <artifactId>worker-framework</artifactId>
-                <version>6.1.0-SNAPSHOT</version>
+                <version>6.1.0-854020-SNAPSHOT</version>
```
Comment (andyreidz) [inline pom.xml]:
```suggestion
                <version>6.1.0-SNAPSHOT</version>
```

File: worker-document/src/main/java/com/hpe/caf/worker/document/tasks/FieldEnrichmentTask.java [hunk 2]
```diff
@@ -108,6 +108,19 @@ protected WorkerResponse handleGeneralFailureImpl(final Throwable failure)
                                   null);
     }
 
+    @Nonnull
+    @Override
+    protected WorkerResponse handlePoisonMessageImpl(final String workerFriendlyName)
+    {
+        return new WorkerResponse(application.getFailureQueue(),
+                TaskStatus.RESULT_EXCEPTION,
+                String.format("'%s' max delivery attempts exceeded.", workerFriendlyName)
```
Comment (andyreidz) [inline worker-document/src/main/java/com/hpe/caf/worker/document/tasks/FieldEnrichmentTask.java]: or
```suggestion
                String.format("'%s' max processing attempts exceeded.", workerFriendlyName)
```

File: worker-document/src/main/java/com/hpe/caf/worker/document/tasks/DocumentTask.java [hunk 3]
```diff
@@ -178,6 +178,19 @@ protected WorkerResponse handleGeneralFailureImpl(final Throwable failure)
         return this.createWorkerResponse();
     }
 
+    @Nonnull
+    @Override
+    protected WorkerResponse handlePoisonMessageImpl(final String workerFriendlyName)
+    {
+        document.getFailures().add(
+                String.format("%s-MAX_DELIVERY_ATTEMPTS_EXCEEDED", workerFriendlyName),
```
Comment (andyreidz) [inline worker-document/src/main/java/com/hpe/caf/worker/document/tasks/DocumentTask.java]: or
```suggestion
                String.format("%s-MAX_PROCESSING_ATTEMPTS_EXCEEDED", workerFriendlyName),
```

File: worker-document/src/main/java/com/hpe/caf/worker/document/tasks/DocumentTask.java [hunk 4]
```diff
@@ -178,6 +178,19 @@ protected WorkerResponse handleGeneralFailureImpl(final Throwable failure)
...
+                String.format("'%s' max delivery attempts exceeded.", workerFriendlyName));
```
Comment (andyreidz) [inline worker-document/src/main/java/com/hpe/caf/worker/document/tasks/DocumentTask.java]:
```suggestion
                String.format("'%s' max processing attempts exceeded.", workerFriendlyName));
```

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**